library/common/config/config.cc

```diff
@@ -386,6 +386,17 @@ R"(
     upstream_connection_options: *upstream_opts
     circuit_breakers: *circuit_breakers_settings
     typed_extension_protocol_options: *h1_protocol_options
+  - name: base_h2_clear
```

Comment:
If we're not including HTTP/3 by default, I think we should also make this something which is configured off but configurable on.  h2c can be really problematic on the internet given middelboxes.

Reply:
This won't get used in any context by default, and will require setting a special header, or more likely (per my previous comment) setting the override via the RequestBuilder. The cluster will exist, however, but be unused with the setup in this PR - is that a concern?

Reply:
Happy to get a second opinion from @RyanTheOptimist but I think if most people don't want it and it adds to config complexity it'd be better to have it not present by default.

Reply:
I agree with @alyssawilk. h2c has the potential to leak secrets onto the wire. Since that is super dangerous and h2c is quite uncommon I would strongly suggest that the bar for using it in Envoy Mobile be has high as possible. (In particular, I think the bar for using h3 should not be higher than the bar for using h2c)

Reply:
If we want to raise the bar, one option would be to require it to be enabled via the EngineBuilder before the cluster is injected. EngineBuilder documentation could come with a prominent warning about usage being risky.

Reply:
worth noting, if I recall correctly h2c frequently doesn't work on the internet - because it's in the clear, all the routing infra that spies on HTTP tries to parse it, and often just kills off connections.  that may have changed since last folks tried but I wouldn't bet on it actually working well in the wild